services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java

```diff
@@ -402,11 +402,17 @@ private List<String> getAllowedInternalSiteCidrs() {
         }
         String[] cidrs = _allowedInternalSites.split(",");
         for (String cidr : cidrs) {
-            if (NetUtils.isValidIp4Cidr(cidr) || NetUtils.isValidIp4(cidr) || !cidr.startsWith("0.0.0.0")) {
-                if (NetUtils.getCleanIp4Cidr(cidr).equals(cidr)) {
+            if (NetUtils.isValidIp4Cidr(cidr)) {
+                if (! NetUtils.getCleanIp4Cidr(cidr).equals(cidr)) {
                     s_logger.warn(String.format("Invalid CIDR %s in %s", cidr, SecStorageAllowedInternalDownloadSites.key()));
                 }
-                allowedCidrs.add(cidr);
+                allowedCidrs.add(NetUtils.getCleanIp4Cidr(cidr));
+            } else if (NetUtils.isValidIp4(cidr)) {
+                String newCidr = cidr + "/32";
+                s_logger.warn(String.format("Ip address is not a valid CIDR; %s using %s/32", cidr, newCidr));
+                allowedCidrs.add(newCidr);
+            } else if (!cidr.startsWith("0.0.0.0")) {
+                allowedCidrs.add(NetUtils.getCleanIp4Cidr(cidr));
```

Comment:
this is what I was confused
I have no idea what is the correct check
maybe it should be `if (cidr.startsWith("0.0.0.0"))` , so that 0.0.0.0/0 is accepted (which is same as `ip route add default via xxx`) ?

the checks need be consistent with

https://github.com/apache/cloudstack/blob/fadb39ece73775e3aadbe715ca428b24e256af34/server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java#L1355-L1362

Reply:
@weizhouapache the global setting states that 0.0.0.0 isn't a valid site:
```
Comma separated list of cidrs internal to the datacenter that can host template download servers, please note 0.0.0.0 is not a valid site
```
I believe that is why that check is in place

Reply:
makes sense.

maybe the check should be
```
if ((NetUtils.isValidIp4Cidr(cidr) 
    || (NetUtils.isValidIp4(cidr) && !cidr.startsWith("0.0.0.0")))
```
?

Reply:
this makes much more sense @weizhouapache . a possible  bug would be to allow `0.1.2.3/0`. not sure if but I'll have a look to see if `NetUtils.getCleanIp4Cidr(cidr)` guards against that.

Reply:
its clean cidr might be `0.0.0.0/0`, which seems to be accepted in elder versions (to be confirmed).